adafruit_tlc5947.py

```diff
@@ -108,12 +108,19 @@ def frequency(self):
         # Must disable a few checks to make pylint happy (ugh).
         # pylint: disable=no-self-use,unused-argument
         @frequency.setter
-        def frequency(self, val):
+        def frequency(self, val: int) -> None:
             raise RuntimeError("Cannot set TLC5947 PWM frequency!")
 
         # pylint: enable=no-self-use,unused-argument
 
-    def __init__(self, spi, latch, *, auto_write=True, num_drivers=1):
+    def __init__(
+        self,
+        spi: "SPI",
+        latch: "DigitalInOut",
```

Comment:
I'm don't think I know the difference (if there is one). But I think for most of the libraries we have been using the object imports where possible instead of strings. So these two types could be:

imports:
```py
try:
    import typing
    from busio import SPI
    from digitalio import DitialInOut
except ImportError:
    pass
```

then for the types:
```py
spi: SPI
latch: DigitalInOut
```

Reply:
Thanks for the feedback. The only problem I face is the `TLC5947` object in the `PWMOut` constructor, because we have a [forward reference](https://peps.python.org/pep-0484/#forward-references). In order to get rid of enclausing the `TLC5947` type in quotes we must use `from __future__ import annotations`. But this line must occur at the beginning of the file, so we can't use a try except block. Should I keep the `TLC5947` type annotation as is or should I add the above line outside of a try except block?

Reply:
Yep, keep the `TLC5947` annotation as it is now. 

That one we have to do it with the string because we can't have that import outside of the try block because it will cause an exception if it runs on the micro controllers.